File: ffupdater/src/main/java/de/marmaro/krt/ffupdater/app/interfaces/Display.java
package de.marmaro.krt.ffupdater.app.interfaces;

import android.content.Context;

import java.util.Optional;

public interface Display {
    String getDisplayTitle(Context context);

    String getDisplayDescription(Context context);

    Optional<String> getDisplayWarning(Context context);

    Optional<String> getDisplayInstalledVersion(Context context);

    String getDisplayDownloadSource(Context context);
}

File: ffupdater/src/main/java/de/marmaro/krt/ffupdater/app/Display.kt
package de.marmaro.krt.ffupdater.app

import android.content.Context

interface Display {
    val displayTitle: Int
    val displayDescription: Int
    val displayWarning: Int?
    fun getDisplayInstalledVersion(context: Context): String
    val displayDownloadSource: Int
    val displayIcon: Int
}